lib/keycard/certificate.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+module Keycard
+  # A Certificate is the central point of information about an authentication
+  # attempt. It logs the verification methods attempted with their statuses
+  # and reports the overall status. When authentication is successful, it holds
+  # the user/account that was verified.
+  class Certificate
```

Comment:
Consider whether a different name might make sense given the pre-existing connotation of certificate as an x509 certificate in this space

Reply:
Yes, I think that is very reasonable. I was struggling with the naming a little... We have three concerns to untangle:

1. The verification methods themselves
2. The abstract notion/name for a verification method
3. The result/log of applying a set of verification methods

I tried to avoid using "method" as an overload of a programming/reserved word, but now believe that would be preferable to a collision in domain terminology. I'm not keen to use the term "strategy", but it might work if we namespaced it so it's clear what the class of strategy is for.

I need to think/discuss this more...

Reply:
Upon some discussion, we decided that AuthenticationResult and AuthenticationMethod (Result and Method in Authentication) are better names. I am working on refactoring this and updating the docs accordingly.